pkg/kubernetes/resources.go

```diff
@@ -82,3 +103,94 @@ func QuotaToMilliCPU(quota, period int64) int64 {
 	}
 	return int64(float64(quota*MilliCPUToCPU)/float64(period) + 0.5)
 }
+
+// MemReqToOomAdj estimates OOM score adjustment based on memory request.
+func MemReqToOomAdj(memRequest int64) int64 {
+	return 1000 - (1000*memRequest)/memCapacity
+}
+
+// OomAdjToMemReq estimates memory request based on OOM score adjustment.
+func OomAdjToMemReq(oomAdj int64, memLimit int64) *int64 {
+	if oomAdj < MinBurstableOOMScoreAdj || oomAdj > MaxBurstableOOMScoreAdj {
+		return nil
+	}
+
+	if req := oomAdjToMemReqEstimates[oomAdj]; req < memLimit || memLimit == 0 {
+		return &req
+	}
+
+	return nil
+}
+
+// CalculateOomAdjToMemReqEstimates calculates a table of memory request estimates
+// for on OOM score adjustment in the range [0, 1000].
+func CalculateOomAdjToMemReqEstimates(memCapacity int64) map[int64]int64 {
+	win := memCapacity / 1000
+	adjToReq := map[int64]int64{}
+
+	for i, req := 0, int64(0); req < memCapacity; i, req = i+1, req+win {
+		adj := MemReqToOomAdj(req)
+		if req == 0 {
+			adjToReq[adj] = req
+			continue
+		}
+
+		lim := req - 10
+		for {
+			if next := MemReqToOomAdj(lim); next != adj {
```

Comment:
Is there a risk that the inner loop starts to last pretty long when system memory capacity is high? Like terabytes?

Reply:
Okay, I updated/fixed the estimate generation. Still a bit ugly, but the unit test, which does a fairly exhaustive checking of calculation in both directions, now passes.